### ISSUE TITLE
windows: various setPrivateNetwork (was: setFirewall) errors

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#716]
To investigate:

* `setFirewall: SetCategory: Exception occurred` 
* `setFirewall: c.Initialize: Incorrect function`
* `setFirewall: winnet.NewNetworkListManager: CoInitialize has not been called`

Tailscale 1.0.4. Windows 10.0.19041.450.

```
$ logprint -i 67ec47bed252768afbe7a40c58c5dbe6bcc4f01f5994e02e1a949c373132859d --since=48h | grep setFirewall
2020-08-24 22:03:09.6754344 +0200 +0200: setFirewall: already category 1
2020-08-25 15:58:57.607225 +0200 +0200: setFirewall: already category 1
2020-08-25 16:25:48.3278826 +0200 +0200: setFirewall: already category 1
2020-08-25 21:55:49.1087509 +0200 +0200: setFirewall: already category 1
2020-08-25 21:55:55.1772661 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:55:56.1870421 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:55:57.1916787 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:55:58.2004544 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:55:59.2269869 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:00.2331471 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:01.25289 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:02.279017 +0200 +0200: setFirewall: c.Initialize: Incorrect function.
2020-08-25 21:56:03.283054 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:04.2861177 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:05.3070837 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:06.3128693 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:07.3185749 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:08.3228492 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:09.3316663 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:10.3462818 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:11.3628387 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:12.3685013 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:56:13.382243 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-25 21:59:39.9754419 +0200 +0200: setFirewall: already category 1
2020-08-25 22:01:52.670637 +0200 +0200: setFirewall: already category 1
2020-08-25 22:01:59.3088941 +0200 +0200: setFirewall: already category 1
2020-08-25 22:02:59.6517569 +0200 +0200: setFirewall: already category 1
2020-08-25 22:03:05.5042097 +0200 +0200: setFirewall: already category 1
2020-08-25 22:03:16.4781264 +0200 +0200: setFirewall: already category 1
2020-08-25 22:04:51.26404 +0200 +0200: setFirewall: already category 1
2020-08-25 22:06:18.5236804 +0200 +0200: setFirewall: already category 1
2020-08-25 22:09:59.5405299 +0200 +0200: setFirewall: already category 1
2020-08-26 08:41:46.2341213 +0200 +0200: setFirewall: already category 1
2020-08-26 13:05:36.2301738 +0200 +0200: setFirewall: already category 1
2020-08-26 13:05:49.0521228 +0200 +0200: setFirewall: already category 1
2020-08-26 13:08:20.1483038 +0200 +0200: setFirewall: already category 1
2020-08-26 13:08:25.3719567 +0200 +0200: setFirewall: already category 1
2020-08-26 14:02:51.0435933 +0200 +0200: setFirewall: already category 1
2020-08-26 14:07:40.4583077 +0200 +0200: setFirewall: already category 1
2020-08-26 14:07:41.1237407 +0200 +0200: setFirewall: already category 1
2020-08-26 14:10:05.3108283 +0200 +0200: setFirewall: already category 1
2020-08-26 14:10:37.432691 +0200 +0200: setFirewall: already category 1
2020-08-26 14:27:40.5820376 +0200 +0200: setFirewall: already category 1
2020-08-26 14:34:11.2401192 +0200 +0200: setFirewall: already category 1
2020-08-26 14:34:18.5663726 +0200 +0200: setFirewall: already category 1
2020-08-26 14:34:22.4686766 +0200 +0200: setFirewall: already category 1
2020-08-26 14:46:23.8751896 +0200 +0200: setFirewall: winnet.NewNetworkListManager: CoInitialize has not been called.
2020-08-26 14:46:24.8765124 +0200 +0200: setFirewall: already category 1
2020-08-26 14:56:41.1595843 +0200 +0200: setFirewall: already category 1
2020-08-26 14:57:17.3656926 +0200 +0200: setFirewall: already category 1
2020-08-26 14:57:29.2879098 +0200 +0200: setFirewall: already category 1
2020-08-26 16:00:50.6482184 +0200 +0200: setFirewall: already category 1
2020-08-26 16:17:58.793211 +0200 +0200: setFirewall: already category 1
2020-08-26 16:23:13.44923 +0200 +0200: setFirewall: already category 1
2020-08-26 16:52:40.4678422 +0200 +0200: setFirewall: already category 1
2020-08-26 16:52:42.9171974 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-26 16:52:43.9387983 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-26 16:52:44.9496076 +0200 +0200: setFirewall: SetCategory: Exception occurred.
2020-08-26 16:52:45.9622186 +0200 +0200: setFirewall: already category 1
2020-08-26 17:15:48.9713861 +0200 +0200: setFirewall: already category 1
2020-08-26 18:56:04.0660893 +0200 +0200: setFirewall: already category 1
2020-08-26 20:48:05.9315743 +0200 +0200: setFirewall: already category 1
2020-08-26 20:50:38.3997455 +0200 +0200: setFirewall: m.GetNetworkConnections: Exception occurred.
2020-08-26 20:50:39.4020574 +0200 +0200: setFirewall: already category 1
```

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2i4gh)

**Comment by bradfitz** [comment]
Note that I renamed `setFirewall` to `setPrivateNetwork` to more accurately reflect what it does.

**Comment by bradfitz** [comment]
In version 1.2.5, some users are still getting errors about this:

```
2020-11-16 20:43:14.4029637 -0500 -0500: setPrivateNetwork(try=5): SetCategory: Exception occurred.
2020-11-16 20:43:15.4070426 -0500 -0500: setPrivateNetwork(try=6): SetCategory: Exception occurred.
2020-11-16 20:43:16.4101211 -0500 -0500: setPrivateNetwork(try=7): SetCategory: Exception occurred.
```

And then the firewall rules we add don't work, because they're scoped to only private interfaces.

**Comment by alexbrainman** [comment]
> * `setFirewall: winnet.NewNetworkListManager: CoInitialize has not been called`

This, perhaps, could happen when using thread without CoInitialize called. If so, this would be fixed by https://github.com/tailscale/tailscale/commit/3b75550ad0b87a2a5e71be3a8ebf737f41391ad6

Alex

**Comment by bradfitz** [comment]
@alexbrainman, oh interesting! That was released in 1.2.7. So maybe that is the problem.

**Comment by bradfitz** [comment]
Still seeing this in 1.4.5.

e.g.

logid 123b4ea602113dea8fc6ab4cfd7dba15aa005ed771f24534e3cf73be965b7254
```
2021-03-08 17:01:26.7399956 -0500 -0500: setPrivateNetwork(try=0): SetCategory: Exception occurred.
2021-03-08 17:01:26.7709967 -0500 -0500: magicsock: ignoring pre-DERP map, STUN-less endpoint update: [[fe80::99d0:ec2d:b2e7:536b]:41641 [fe80::ccd6:f6f5:f15c:651e]:41641 10.0.5.5:41641]
2021-03-08 17:01:26.8719972 -0500 -0500: LinkChange: minor
2021-03-08 17:01:26.8729976 -0500 -0500: magicsock: starting endpoint update (link-change-minor)
2021-03-08 17:01:26.9479988 -0500 -0500: magicsock: ignoring pre-DERP map, STUN-less endpoint update: [[fe80::99d0:ec2d:b2e7:536b]:41641 [fe80::ccd6:f6f5:f15c:651e]:41641 10.0.5.5:41641]
2021-03-08 17:01:27.0220007 -0500 -0500: ipnserver: conn1: incoming control connection
2021-03-08 17:01:27.0230013 -0500 -0500: Start: Prefs{ra=true dns=true want=true Persist=nil}
2021-03-08 17:01:27.0590011 -0500 -0500: using frontend prefs: Prefs{ra=true dns=true want=true Persist=nil}
2021-03-08 17:01:27.0590011 -0500 -0500: generating new machine key
2021-03-08 17:01:27.0680007 -0500 -0500: router: firewall: cleared Tailscale-In firewall rules in 466ms
2021-03-08 17:01:27.0680007 -0500 -0500: router: firewall: deleting any prior Tailscale-Process rule...
2021-03-08 17:01:27.1300031 -0500 -0500: machine key written to store
2021-03-08 17:01:27.1300031 -0500 -0500: Start: serverMode=false
2021-03-08 17:01:27.1300031 -0500 -0500: netmap packet filter: (not ready yet)
2021-03-08 17:01:27.1300031 -0500 -0500: got initial portlist info in 0s
2021-03-08 17:01:27.1300031 -0500 -0500: magicsock: disco key = d:8ade273278544a62
2021-03-08 17:01:27.1300031 -0500 -0500: control: HostInfo: {"IPNVersion":"1.4.5-tb89c65304-gf09ac28c2","FrontendLogID":"71becc0806e438b3b1dfd7f9ef775ac38260ff477ffdb97aeda436bce0006467","BackendLogID":"123b4ea602113dea8fc6ab4cfd7dba15aa005ed771f24534e3cf73be965b7254","OS":"windows","OSVersion":"10.0.17763.1757","Hostname":"ND-AZFA01","GoArch":"amd64","Services":[{"Proto":"tcp","Port":3389,"Description":"TermService"}]}
2021-03-08 17:01:27.1300031 -0500 -0500: Backend: logs: be:123b4ea602113dea8fc6ab4cfd7dba15aa005ed771f24534e3cf73be965b7254 fe:71becc0806e438b3b1dfd7f9ef775ac38260ff477ffdb97aeda436bce0006467
2021-03-08 17:01:27.1300031 -0500 -0500: control: client.Login(false, 0)
2021-03-08 17:01:27.1300031 -0500 -0500: control: authRoutine: state:new; wantLoggedIn=true
2021-03-08 17:01:27.1300031 -0500 -0500: control: direct.TryLogin(token=false, flags=0)
2021-03-08 17:01:27.1300031 -0500 -0500: control: doLogin(regen=false, hasUrl=false)
2021-03-08 17:01:27.1320018 -0500 -0500: control: mapRoutine: state:authenticating
2021-03-08 17:01:27.1560045 -0500 -0500: router: firewall: adding Tailscale-Process rule to allow UDP for "C:\\Program Files (x86)\\Tailscale IPN\\tailscale-ipn.exe" ...
2021-03-08 17:01:27.257003 -0500 -0500: router: firewall: added Tailscale-Process rule in 101ms
2021-03-08 17:01:27.257003 -0500 -0500: router: firewall: ending netsh goroutine
2021-03-08 17:01:27.5420056 -0500 -0500: control: Generating a new nodekey.
2021-03-08 17:01:27.5420056 -0500 -0500: control: RegisterReq: onode=[AAAAA] node=[YZNtr] fup=false
2021-03-08 17:01:27.6400075 -0500 -0500: control: RegisterReq: got response; nodeKeyExpired=false, machineAuthorized=false; authURL=true
2021-03-08 17:01:27.6400075 -0500 -0500: control: AuthURL is https://login.tailscale.com/a/1348a9c12663
2021-03-08 17:01:27.6400075 -0500 -0500: control: sendStatus: authRoutine-url: state:url-visit-required
2021-03-08 17:01:27.6400075 -0500 -0500: Received auth URL: https://login.tailsc...
2021-03-08 17:01:27.6400075 -0500 -0500: Switching ipn state NoState -> NeedsLogin (WantRunning=true)
2021-03-08 17:01:27.6400075 -0500 -0500: blockEngineUpdates(true)
2021-03-08 17:01:27.6410098 -0500 -0500: wgengine: Reconfig: configuring userspace wireguard config (with 0/0 peers)
2021-03-08 17:01:27.6410098 -0500 -0500: UDP bind update skipped
2021-03-08 17:01:27.6410098 -0500 -0500: wgengine: Reconfig: configuring router
2021-03-08 17:01:27.6410098 -0500 -0500: router: firewall: marking for removal
2021-03-08 17:01:27.6410098 -0500 -0500: router: firewall: starting netsh goroutine
2021-03-08 17:01:27.6410098 -0500 -0500: router: firewall: ending netsh goroutine
2021-03-08 17:01:27.6600113 -0500 -0500: wgengine: Reconfig done
2021-03-08 17:01:27.6600113 -0500 -0500: authReconfig: blocked, skipping.
2021-03-08 17:01:27.6600113 -0500 -0500: control: authRoutine: state:url-visit-required; wantLoggedIn=true
2021-03-08 17:01:27.6600113 -0500 -0500: control: direct.WaitLoginURL
2021-03-08 17:01:27.6600113 -0500 -0500: control: doLogin(regen=false, hasUrl=true)
2021-03-08 17:01:27.6600113 -0500 -0500: control: RegisterReq: onode=[AAAAA] node=[YZNtr] fup=true
2021-03-08 17:01:27.6600113 -0500 -0500: setPrivateNetwork(try=0): SetCategory: Exception occurred.
2021-03-08 17:01:27.7460104 -0500 -0500: setPrivateNetwork(try=1): SetCategory: Exception occurred.
2021-03-08 17:01:28.6673861 -0500 -0500: setPrivateNetwork(try=1): SetCategory: Exception occurred.
2021-03-08 17:01:28.7525728 -0500 -0500: setPrivateNetwork(try=2): SetCategory: Exception occurred.
2021-03-08 17:01:29.6821823 -0500 -0500: setPrivateNetwork(try=2): SetCategory: Exception occurred.
2021-03-08 17:01:29.7261819 -0500 -0500: monitor: interface state changed (on poll)
2021-03-08 17:01:29.7421819 -0500 -0500: LinkChange: minor
2021-03-08 17:01:29.7421819 -0500 -0500: magicsock: starting endpoint update (link-change-minor)
2021-03-08 17:01:29.7511821 -0500 -0500: magicsock: ignoring pre-DERP map, STUN-less endpoint update: [[fe80::99d0:ec2d:b2e7:536b]:41641 [fe80::ccd6:f6f5:f15c:651e]:41641 10.0.5.5:41641]
2021-03-08 17:01:29.7611841 -0500 -0500: setPrivateNetwork(try=3): SetCategory: Exception occurred.
2021-03-08 17:01:30.6915468 -0500 -0500: setPrivateNetwork(try=3): SetCategory: Exception occurred.
2021-03-08 17:01:30.7753032 -0500 -0500: setPrivateNetwork(try=4): SetCategory: Exception occurred.
2021-03-08 17:01:31.7068643 -0500 -0500: setPrivateNetwork(try=4): SetCategory: Exception occurred.
2021-03-08 17:01:31.78804 -0500 -0500: setPrivateNetwork(try=5): SetCategory: Exception occurred.
2021-03-08 17:01:32.4124604 -0500 -0500: monitor: got windows change event after 5.7023441s: evt=addr-o
2021-03-08 17:01:32.4194602 -0500 -0500: monitor: got windows change event after 5.0418ms: evt=addr-o
2021-03-08 17:01:32.4502953 -0500 -0500: LinkChange: minor
2021-03-08 17:01:32.4502953 -0500 -0500: magicsock: starting endpoint update (link-change-minor)
2021-03-08 17:01:32.4562955 -0500 -0500: magicsock: ignoring pre-DERP map, STUN-less endpoint update: [[fe80::99d0:ec2d:b2e7:536b]:41641 [fe80::ccd6:f6f5:f15c:651e]:41641 10.0.5.5:41641]
2021-03-08 17:01:32.5750345 -0500 -0500: LinkChange: minor
2021-03-08 17:01:32.5750345 -0500 -0500: magicsock: starting endpoint update (link-change-minor)
2021-03-08 17:01:32.5820373 -0500 -0500: magicsock: ignoring pre-DERP map, STUN-less endpoint update: [[fe80::99d0:ec2d:b2e7:536b]:41641 [fe80::ccd6:f6f5:f15c:651e]:41641 10.0.5.5:41641]
2021-03-08 17:01:32.730815 -0500 -0500: setPrivateNetwork(try=5): SetCategory: Exception occurred.
2021-03-08 17:01:32.7983826 -0500 -0500: setPrivateNetwork(try=6): SetCategory: Exception occurred.
2021-03-08 17:01:33.7366337 -0500 -0500: setPrivateNetwork(try=6): SetCategory: Exception occurred.
2021-03-08 17:01:33.8048988 -0500 -0500: setPrivateNetwork(try=7): SetCategory: Exception occurred.
2021-03-08 17:01:34.7446805 -0500 -0500: setPrivateNetwork(try=7): SetCategory: Exception occurred.
2021-03-08 17:01:34.8134741 -0500 -0500: setPrivateNetwork(try=8): SetCategory: Exception occurred.
2021-03-08 17:01:35.4485634 -0500 -0500: monitor: interface state changed (on poll)
2021-03-08 17:01:35.4634979 -0500 -0500: LinkChange: minor
2021-03-08 17:01:35.4634979 -0500 -0500: magicsock: starting endpoint update (link-change-minor)
2021-03-08 17:01:35.4734965 -0500 -0500: magicsock: ignoring pre-DERP map, STUN-less endpoint update: [[fe80::99d0:ec2d:b2e7:536b]:41641 [fe80::ccd6:f6f5:f15c:651e]:41641 10.0.5.5:41641]
2021-03-08 17:01:35.7524975 -0500 -0500: setPrivateNetwork(try=8): SetCategory: Exception occurred.
2021-03-08 17:01:35.8209207 -0500 -0500: setPrivateNetwork(try=9): SetCategory: Exception occurred.
2021-03-08 17:01:35.9082467 -0500 -0500: monitor: got windows change event after 3.4907434s: evt=route-o
2021-03-08 17:01:35.9332476 -0500 -0500: monitor: got windows change event after 24ms: evt=addr-o
2021-03-08 17:01:35.9670311 -0500 -0500: LinkChange: minor
2021-03-08 17:01:35.9670311 -0500 -0500: magicsock: starting endpoint update (link-change-minor)
2021-03-08 17:01:35.9730337 -0500 -0500: magicsock: ignoring pre-DERP map, STUN-less endpoint update: [[fe80::99d0:ec2d:b2e7:536b]:41641 [fe80::ccd6:f6f5:f15c:651e]:41641 10.0.5.5:41641]
2021-03-08 17:01:36.0894716 -0500 -0500: LinkChange: minor
2021-03-08 17:01:36.0894716 -0500 -0500: magicsock: starting endpoint update (link-change-minor)
2021-03-08 17:01:36.0964734 -0500 -0500: magicsock: ignoring pre-DERP map, STUN-less endpoint update: [[fe80::99d0:ec2d:b2e7:536b]:41641 [fe80::ccd6:f6f5:f15c:651e]:41641 10.0.5.5:41641]
2021-03-08 17:01:36.7620214 -0500 -0500: setPrivateNetwork(try=9): SetCategory: Exception occurred.
2021-03-08 17:01:36.8369295 -0500 -0500: setPrivateNetwork(try=10): SetCategory: Exception occurred.
2021-03-08 17:01:37.2341964 -0500 -0500: control: HostInfo: {"IPNVersion":"1.4.5-tb89c65304-gf09ac28c2","FrontendLogID":"71becc0806e438b3b1dfd7f9ef775ac38260ff477ffdb97aeda436bce0006467","BackendLogID":"123b4ea602113dea8fc6ab4cfd7dba15aa005ed771f24534e3cf73be965b7254","OS":"windows","OSVersion":"10.0.17763.1757","Hostname":"ND-AZFA01","GoArch":"amd64"}
2021-03-08 17:01:37.2341964 -0500 -0500: control: Hostinfo: &{1.4.5-tb89c65304-gf09ac28c2 71becc0806e438b3b1dfd7f9ef775ac38260ff477ffdb97aeda436bce0006467 123b4ea602113dea8fc6ab4cfd7dba15aa005ed771f24534e3cf73be965b7254 windows 10.0.17763.1757   ND-AZFA01 false false amd64 [] [] [] NetInfo(nil)}
2021-03-08 17:01:37.2341964 -0500 -0500: control: cancelMapSafely: synced=false
2021-03-08 17:01:37.2341964 -0500 -0500: control: cancelMapSafely: wrote to channel
2021-03-08 17:01:37.2341964 -0500 -0500: control: mapRoutine: new map needed while idle.
2021-03-08 17:01:37.2341964 -0500 -0500: control: mapRoutine: state:url-visit-required
2021-03-08 17:01:37.774137 -0500 -0500: setPrivateNetwork(try=10): SetCategory: Exception occurred.
2021-03-08 17:01:37.8576889 -0500 -0500: setPrivateNetwork(try=11): SetCategory: Exception occurred.
2021-03-08 17:01:38.8011456 -0500 -0500: setPrivateNetwork(try=11): success
2021-03-08 17:01:38.8854832 -0500 -0500: setPrivateNetwork(try=12): success
```

It did eventually work. Do we need to wait for something, @peske?

Also, were two running concurrently? What's with the repeated try numbers:

```
2021-03-08 17:01:32.7983826 -0500 -0500: setPrivateNetwork(try=6): SetCategory: Exception occurred.
2021-03-08 17:01:33.7366337 -0500 -0500: setPrivateNetwork(try=6): SetCategory: Exception occurred.
2021-03-08 17:01:33.8048988 -0500 -0500: setPrivateNetwork(try=7): SetCategory: Exception occurred.
2021-03-08 17:01:34.7446805 -0500 -0500: setPrivateNetwork(try=7): SetCategory: Exception occurred.
```
?

**Comment by bradfitz** [comment]
Still seeing this in 1.4.6.

Customer issue: https://app.frontapp.com/open/cnv_er94dg1
Logs: 51f06537796dbf8e5d4168466b2982086adca4b5e75a125dbc03888dec0ba462
```
2021-03-15 10:08:14.3773208 +1100 +1100: Program starting: v1.4.6-t937166517-gc4daba224: []string{"C:\\Program Files (x86)\\Tailscale IPN\\tailscale-ipn.exe", "/subproc", "51f06537796dbf8e5d4168466b2982086adca4b5e75a125dbc03888dec0ba462"}
...
2021-03-15 10:08:32.2989746 +1100 +1100: setPrivateNetwork(try=17): SetCategory: Exception occurred.
2021-03-15 10:08:33.3177151 +1100 +1100: setPrivateNetwork(try=18): SetCategory: Exception occurred.
2021-03-15 10:08:34.3360195 +1100 +1100: setPrivateNetwork(try=19): SetCategory: Exception occurred.
2021-03-15 10:08:35.336036 +1100 +1100: setPrivateNetwork: adapter LUID 14918723521478656 not found after 20 tries, giving up
```

The Windows version there is 10.0.17763.1158 ("Windows 10 October 2018 Update", aka "Redstone 5").

Maybe it's specific to the Windows version?

I guess we need some old Windows images. I read that https://rufus.ie/ lets you download old installers from Microsoft's site.

**Comment by DentonGentry** [comment]
I believe this was fixed with the Windows firewall revamps in 1.6 while adding IPv6 support:
```
% ./logprint -i d4c640b060b6c6d00db97e9ccde197face73efef992442ef896aefd72577ccce -since=240h | grep "Exception occurred."
collection: tailnode.log.tailscale.io, logid: d4c640b060b6c6d00db97e9ccde197face73efef992442ef896aefd72577ccce
% ./logprint -i c17896a97ebd44dc6451f858f064a6db45ec3c4289dc2b196aa1490a9e0195f2 -since=240h | grep "Exception occurred."
collection: tailnode.log.tailscale.io, logid: c17896a97ebd44dc6451f858f064a6db45ec3c4289dc2b196aa1490a9e0195f2
%
```
